src/utils.h

```diff
@@ -28,3 +32,30 @@ limitations under the License.
                   << std::endl;                                                \
         abort();                                                               \
     } while (false)
+
+// According to the agreement, Random Value Generator is the only way to get any random value in YARPGen. It is used for different random decisions all over the source code.
+class RandValGen {
+  public:
+    // Specific seed can be passed to constructor to reproduce the test.
+    // Zero value is reserved (it notifies RandValGen that it can choose any)
+    RandValGen(uint64_t _seed);
+
+    template <typename T> T getRandValue(T from, T to) {
+        // Using long long instead of T is a hack.
+        // getRandValue is used with all kind of integer types, including chars. While standard is not allowing it to be used with uniform_int_distribution<> algorithm. Though, clang and gcc ok with it, but VS doesn't compile such code. For details see C++17, $26.5.1.1e [rand.req.genl]. This issue is also discussed in issue 2326 (closed as not a defect and reopened as feature request N4296).
+        std::uniform_int_distribution<long long> dis(from, to);
+        return dis(rand_gen);
+    }
+
+  private:
+    uint64_t seed;
+    std::mt19937_64 rand_gen;
+};
+
+template <> inline bool RandValGen::getRandValue<bool>(bool from, bool to) {
```

Comment:
Does it make sense to accept any parameters for bool version?

Reply:
It is explicit template instantiation for function. Let's keep them uniform.